pyproject.toml

```diff
@@ -0,0 +1,68 @@
+[build-system]
+requires = ["setuptools", "Cython"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools.packages.find]
+where = ["."]
+exclude = ["sphinx*"]
+
+[tool.setuptools.package-data]
+"*" = ["*.json", "*.dat"]
+"omas.samples" = ["*"]
+"omas"=["*.pyx"]
+
+[tool.setuptools.dynamic]
+version = {file = "omas/version"}
+
+[project]
+name = 'omas'
+dynamic = ["version"]
+description = 'Ordered Multidimensional Array Structures'
+authors = [
+    {name = 'Orso Meneghini', email = 'meneghini@fusion.gat.com'},
+]
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3 :: Only',
+    'License :: OSI Approved :: MIT License',
+]
+keywords = ["integrated modeling",  "OMFIT", "IMAS",  "ITER"]
+
+requires-python = '>=3.8'
```

Comment:
It would require bumping this from 3.6 to 3.8. 
3.6 was end of life in 2021, so I hope that this is ok?

Reply:
OMFIT is at python=3.7 and uses omas as an integral part.  What changes are you making here that require python>=3.8?

Reply:
I looked at what the recent versions of setuptools require.
But what would be needed here is support for [PEP-621](https://peps.python.org/pep-0621/) which looks like it landed in `setuptools` version [v61-0-0](https://setuptools.pypa.io/en/latest/history.html#v61-0-0) which still supports >=3.7. 

I could try and see giving if it works for 3.7.

Reply:
@smithsp I double checked and I was correct that in theory `3.7` should have been sufficient to enable usage of `pyproject.toml` based package configurations.

However, the problem here is the dependency on `pygacode` which is forced to version `0.71` from 2022 because that's the last version that supports `3.7`, and that version does not correctly specify its build dependencies (requires numpy). This is fixed in for versions `>1.0` but those require python `>=3.8` which is why the build here works for `3.8` but nor for `3.7`.

Reply:
We are actively working on migrating OMFIT to a later python version.